src/main/java/com/squareup/javawriter/ClassName.java

```diff
@@ -103,42 +102,6 @@ public String canonicalName() {
     return fullyQualifiedName;
   }
 
-  public String classFileName() {
-    StringBuilder builder = new StringBuilder();
-    Joiner.on('$').appendTo(builder, enclosingSimpleNames());
-    if (!enclosingSimpleNames().isEmpty()) {
-      builder.append('$');
-    }
-    return builder.append(simpleName()).toString();
-  }
-
-  public ClassName topLevelClassName() {
-    Iterator<String> enclosingIterator = enclosingSimpleNames().iterator();
-    return enclosingIterator.hasNext()
-        ? new ClassName(packageName(), ImmutableList.<String>of(),
-            enclosingIterator.next())
-        : this;
-  }
-
-  public ClassName nestedClassNamed(String memberClassName) {
-    checkNotNull(memberClassName);
-    checkArgument(SourceVersion.isIdentifier(memberClassName));
-    checkArgument(Ascii.isUpperCase(memberClassName.charAt(0)));
-    return new ClassName(packageName(),
-        new ImmutableList.Builder<String>()
-            .addAll(enclosingSimpleNames())
-            .add(simpleName())
-            .build(),
-        memberClassName);
-  }
-
-  public ClassName peerNamed(String peerClassName) {
-    checkNotNull(peerClassName);
-    checkArgument(SourceVersion.isIdentifier(peerClassName));
-    checkArgument(Ascii.isUpperCase(peerClassName.charAt(0)));
```

Comment:
... these methods are all pretty suspicious anyway. This breaks for `_Remote_Stub`, a foolishly-named JDK class.